chapters/slang.adoc

```diff
@@ -0,0 +1,744 @@
+
+// Set entry point
+spAddEntryPoint(compileRequest, translationUnitIndex, "main", SLANG_STAGE_FRAGMENT);
+
+// Compile
+int compileResult = spCompile(compileRequest);
+if (compileResult != 0)
+{
+    const char* diagnostics = spGetDiagnosticOutput(compileRequest);
+    // Handle compilation error
+}
+
+// Get compiled code
+size_t codeSize = 0;
+const void* codeData = spGetEntryPointCode(compileRequest, 0, &codeSize);
+
+// Create Vulkan shader module
+VkShaderModuleCreateInfo shaderModuleCI{};
+shaderModuleCI.sType = VK_STRUCTURE_TYPE_SHADER_MODULE_CREATE_INFO;
+shaderModuleCI.codeSize = codeSize;
+shaderModuleCI.pCode = static_cast<const uint32_t*>(codeData);
+VkShaderModule shaderModule;
+vkCreateShaderModule(device, &shaderModuleCI, nullptr, &shaderModule);
+----
+
+=== Slang-specific features for Vulkan
+
+When using Slang with Vulkan, you can take advantage of several Slang-specific features:
+
+* **Cross-compilation**: Slang can target multiple backends from the same source code
+* **Shader parameters**: Slang provides a more structured way to declare shader parameters
+* **Resource binding**: Slang offers more flexible resource binding options
+* **Reflection**: Slang provides rich reflection capabilities for shader introspection
+
+Example of Slang shader parameters:
+
+[source, slang]
+----
+// Define a parameter block for Vulkan
+[[vk::binding(0, 0)]]
+ParameterBlock<SceneData> sceneData;
+
+// Define a struct for scene data
+struct SceneData
+{
+    float4x4 viewProj;
+    float3 cameraPosition;
+    float time;
+    Texture2D diffuseMap;
+    SamplerState samplerState;
+};
+----
+
+== Best Practices for Writing Shaders in Slang for Vulkan
+
+=== Code Organization
+
+* *Use modules*: Organize code into logical modules with clear responsibilities
+* *Leverage imports*: Use imports instead of includes for better dependency management
+* *Separate shader stages*: Keep different shader stages in separate files
+* *Use interfaces*: Define interfaces for common functionality to enable polymorphism
+
+Example of a well-organized Slang shader:
+
+[source,slang]
+----
+module Rendering.PBR;
+
+import Rendering.Common;
+import Rendering.Lighting;
+
+export interface IMaterial {
+    float3 evaluateBRDF(float3 viewDir, float3 lightDir, float3 normal);
+}
+
+export struct PBRMaterial : IMaterial {
+    float3 albedo;
+    float roughness;
+    float metallic;
+
+    float3 evaluateBRDF(float3 viewDir, float3 lightDir, float3 normal) {
+        // PBR implementation
+        return calculatePBR(albedo, roughness, metallic, viewDir, lightDir, normal);
+    }
+}
+
+[shader("fragment")]
+export float4 fragmentMain(VSOutput input) {
+    // Shader implementation
+}
+----
+
+=== Performance Considerations
+
+* *Use generics judiciously*: While powerful, excessive use of generics can increase compile times
+* *Profile interface dispatch*: Test performance impact of interface-based polymorphism
```

Comment:
Another place where mentioning the flag that will hopefully get added in https://github.com/shader-slang/slang/pull/9273 would be useful.